ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -92,6 +99,192 @@ protected override BorderedCanvas CreateViewInstance(ThemedReactContext reactCon
             return new BorderedCanvas();
         }
 
+        #region drag and drop
+#if WINDOWS_UWP
+
+        /// <summary>
+        /// Enables the Canvas as a drop target.
+        /// </summary>
+        [ReactProp("allowDrop")]
+        public void SetAllowDrop(BorderedCanvas view, bool allowDrop)
+        {
+            view.AllowDrop = allowDrop;
+
+            if (allowDrop)
+            {
+                view.DragEnter += OnDragEnter;
+                view.DragOver += OnDragOver;
+                view.Drop += OnDrop;
+                view.DragLeave += OnDragLeave;
+
+            }
+            else
+            {
+                view.DragEnter -= OnDragEnter;
+                view.DragOver -= OnDragOver;
+                view.Drop -= OnDrop;
+                view.DragLeave -= OnDragLeave;
+            }
+        }
+
+        private async Task<JObject> GetDataTransferInfo(DataPackageView data, bool drop = false)
+        {
+            var files = new JArray();
+            var items = new JArray();
+            var types = new JArray();
+
+            if (data.Contains(StandardDataFormats.StorageItems))
+            {
+                foreach (var item in await data.GetStorageItemsAsync())
+                {
+                    try
+                    {
+                        var file = item as StorageFile;
+
+                        if (file != null)
+                        {
+                            var props = await file.GetBasicPropertiesAsync();
+                            var type = file.ContentType;
+                            var path = drop ? "file://cache/" + GetAccessToken(file) : "";
```

Comment:
>"file://cache/" + GetAccessToken(file) [](start = 46, length = 38)

Are there better APIs for creating URIs than string concatenation?

Reply:
It would make sense to use a special API if we weren't 100% sure about parts of the URI we're merging together. In this case the structure is simple and is in our control: scheme prefix + GUID. No special symbols, no user input, no unknowns. For this reason I think that a basic concatenation is good enough here.